In [2]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_excel('Creatives.xlsx')
df

Position  Number of employees  \
0                              Manager                    1   
1                         Video editor                    1   
2                Video editor(Interns)                    5   
3      Contingency team(Video Editing)                    3   
4                     Content designer                    1   
5             Content designer(Intern)                    5   
6  Contingency team(Content Designing)                    8   
7                       Content writer                    1   
8               Content writer(Intern)                    5   
9    Contingency team(Content writing)                    4   

   Salary/month/employee  Salary/month  Salary/year/employee  Salary/year  \
0               27916.67      27916.67                335000       335000   
1               20000.00      20000.00                240000       240000   
2                1000.00       5000.00                 12000        60000   
3                1000.00       3000.00                 12000        36000   
4               20833.33      20833.33                250000       250000   
5                1000.00       5000.00                 12000        60000   
6                1000.00       8000.00                 12000        96000   
7               10000.00      10000.00                120000       120000   
8                1000.00       5000.00                 12000        60000   
9                1000.00       4000.00                 12000        48000   

   Month 1  Month 2  Month 3  Month 4  ...  Month 16  Month 17  Month 18  \
0    27917    27917    83750    83750  ...    223333    223333    251250   
1    20000    20000    60000    60000  ...    140000    160000    160000   
2     5000     5000    15000    15000  ...     35000     40000     40000   
3     3000     3000     9000     9000  ...     21000     24000     24000   
4    20833    20833    62500    62500  ...    187500    187500    187500   
5     5000     5000    15000    15000  ...     45000     45000     45000   
6     8000     8000    24000    24000  ...     72000     72000     72000   
7    10000    10000    80000    80000  ...    230000    230000    240000   
8     5000     5000    40000    40000  ...    115000    115000    120000   
9     4000     4000    32000    32000  ...     92000     92000     96000   

   Month 19  Month 20  Month 21  Month 22  Month 23  Month 24  \
0    251250    251250    279167    279167    279167    279167   
1    160000    160000    180000    180000    180000    180000   
2     40000     40000     45000     45000     45000     45000   
3     24000     24000     27000     27000     27000     27000   
4    208333    208333    208333    229167    229167    229167   
5     50000     50000     50000     55000     55000     55000   
6     80000     80000     80000     88000     88000     88000   
7    250000    260000    270000    270000    280000    290000   
8    125000    130000    135000    135000    140000    145000   
9    100000    104000    108000    108000    112000    116000   

                       Unnamed: 30  
0                              NaN  
1                              NaN  
2         Starting salary - 2.5LPA  
3  Initially, we'll hire 5 interns  
4                              NaN  
5         Starting salary - 1.2LPA  
6  Initially, we'll hire 5 interns  
7         Starting salary - 1.2LPA  
8  Initially, we'll hire 5 interns  
9                              NaN  

[10 rows x 31 columns]

In [6]:
df.describe()

Number of employees  Salary/month/employee  Salary/month  \
count            10.000000              10.000000     10.000000   
mean              3.400000            8475.000000  10875.000000   
std               2.412928           10546.783885   8785.795688   
min               1.000000            1000.000000   3000.000000   
25%               1.000000            1000.000000   5000.000000   
50%               3.500000            1000.000000   6500.000000   
75%               5.000000           17500.000000  17500.000000   
max               8.000000           27916.670000  27916.670000   

       Salary/year/employee    Salary/year       Month 1       Month 2  \
count              10.00000      10.000000     10.000000     10.000000   
mean           101700.00000  130500.000000  10875.000000  10875.000000   
std            126561.40363  105429.544668   8785.825251   8785.825251   
min             12000.00000   36000.000000   3000.000000   3000.000000   
25%             12000.00000   60000.000000   5000.000000   5000.000000   
50%             12000.00000   78000.000000   6500.000000   6500.000000   
75%            210000.00000  210000.000000  17500.000000  17500.000000   
max            335000.00000  335000.000000  27917.000000  27917.000000   

            Month 3       Month 4        Month 5  ...       Month 15  \
count     10.000000     10.000000      10.000000  ...      10.000000   
mean   42125.000000  42125.000000   52100.000000  ...  114183.300000   
std    27701.095546  27701.095546   36055.216571  ...   75553.232816   
min     9000.000000   9000.000000    9000.000000  ...   21000.000000   
25%    17250.000000  17250.000000   23000.000000  ...   51750.000000   
50%    36000.000000  36000.000000   45000.000000  ...   99000.000000   
75%    61875.000000  61875.000000   77499.750000  ...  175625.000000   
max    83750.000000  83750.000000  111667.000000  ...  223333.000000   

            Month 16       Month 17     Month 18       Month 19  \
count      10.000000      10.000000      10.0000      10.000000   
mean   116083.300000  118883.300000  123575.0000  128858.300000   
std     76988.132924   76938.278798   82849.2649   85170.147744   
min     21000.000000   24000.000000   24000.0000   24000.000000   
25%     51750.000000   51750.000000   51750.0000   57500.000000   
50%    103500.000000  103500.000000  108000.0000  112500.000000   
75%    175625.000000  180625.000000  180625.0000  196249.750000   
max    230000.000000  230000.000000  251250.0000  251250.000000   

            Month 20       Month 21       Month 22       Month 23  \
count      10.000000      10.000000      10.000000      10.000000   
mean   130758.300000  138250.000000  141633.400000  143533.400000   
std     86630.679195   92525.808879   93430.620618   94809.961163   
min     24000.000000   27000.000000   27000.000000   27000.000000   
25%     57500.000000   57500.000000   63250.000000   63250.000000   
50%    117000.000000  121500.000000  121500.000000  126000.000000   
75%    196249.750000  201249.750000  216875.250000  216875.250000   
max    260000.000000  279167.000000  279167.000000  280000.000000   

           Month 24  
count      10.00000  
mean   145433.40000  
std     96290.64188  
min     27000.00000  
25%     63250.00000  
50%    130500.00000  
75%    216875.25000  
max    290000.00000  

[8 rows x 29 columns]

In [7]:
num_features = [feature for feature in df.columns if df[feature].dtype!='O']
for feature in num_features:
    print(f"{feature} - {df[feature].sum()}")

Number of employees - 34
Salary/month/employee - 84750.0
Salary/month - 108750.0
Salary/year/employee - 1017000
Salary/year - 1305000
Month 1 - 108750
Month 2 - 108750
Month 3 - 421250
Month 4 - 421250
Month 5 - 521000
Month 6 - 601834
Month 7 - 667750
Month 8 - 748583
Month 9 - 814500
Month 10 - 895333
Month 11 - 942250
Month 12 - 961250
Month 13 - 1042084
Month 14 - 1089000
Month 15 - 1141833
Month 16 - 1160833
Month 17 - 1188833
Month 18 - 1235750
Month 19 - 1288583
Month 20 - 1307583
Month 21 - 1382500
Month 22 - 1416334
Month 23 - 1435334
Month 24 - 1454334


In [8]:
df['Month 12'][df['Position']=='Manager']

0    195417
Name: Month 12, dtype: int64

This analysis can be helpful in part where we can show after a particular number of months what is the salary and what is the total amount being spent on creatives dept 

Eg. For Month 12:

Total Amount being spent on salary for creatives dept is 961250.

and The Salary of Manger in month 12 is 195417. 

Interactivity also can be added using component datepicker or rangeslider or input box where we have months from 1 to 24 and there is pie chart which visualizes for a month for the salary is divided for each position of creatives dept

In [16]:
df1 = pd.read_excel('HR.xlsx')
df1

Position  Number of employees  Salary/month/employee  Salary/month  \
0  HR manager                    1               25000.00         25000   
1   Recruiter                    3                9166.67         27500   

   Salary/year/employee  Salary/year  Month 1  Month 2  Month 3  Month 4  ...  \
0                300000       300000    25000    25000    25000    25000  ...   
1                110000       330000    27500    27500    27500    27500  ...   

   Month 15  Month 16  Month 17  Month 18  Month 19  Month 20  Month 21  \
0     25000     25000     25000     25000     25000     25000     25000   
1     27500     27500     27500     27500     27500     27500     27500   

   Month 22  Month 23  Month 24  
0     25000     25000     25000  
1     27500     27500     27500  

[2 rows x 30 columns]

In [17]:
for feature in num_features:
    print(f"{feature} - {df1[feature].sum()}")

Number of employees - 4
Salary/month/employee - 34166.67
Salary/month - 52500
Salary/year/employee - 410000
Salary/year - 630000
Month 1 - 52500
Month 2 - 52500
Month 3 - 52500
Month 4 - 52500
Month 5 - 52500
Month 6 - 52500
Month 7 - 52500
Month 8 - 52500
Month 9 - 52500
Month 10 - 52500
Month 11 - 52500
Month 12 - 52500
Month 13 - 52500
Month 14 - 52500
Month 15 - 52500
Month 16 - 52500
Month 17 - 52500
Month 18 - 52500
Month 19 - 52500
Month 20 - 52500
Month 21 - 52500
Month 22 - 52500
Month 23 - 52500
Month 24 - 52500


Here this is a problem that the salary amount is same for each year so there can the number will stay constant throughout

In [3]:
df3 = pd.read_excel('CombinedData.xlsx')
df3

Position  Number of employees  \
0                               Manager                    1   
1                          Video editor                    1   
2                 Video editor(Interns)                    5   
3       Contingency team(Video Editing)                    3   
4                      Content designer                    1   
5              Content designer(Intern)                    5   
6   Contingency team(Content Designing)                    8   
7                        Content writer                    1   
8                Content writer(Intern)                    5   
9     Contingency team(Content writing)                    4   
10                           HR manager                    1   
11                            Recruiter                    3   

    Salary/month/employee  Salary/month  Salary/year/employee  Salary/year  \
0                27916.67      27916.67                335000       335000   
1                20000.00      20000.00                240000       240000   
2                 1000.00       5000.00                 12000        60000   
3                 1000.00       3000.00                 12000        36000   
4                20833.33      20833.33                250000       250000   
5                 1000.00       5000.00                 12000        60000   
6                 1000.00       8000.00                 12000        96000   
7                10000.00      10000.00                120000       120000   
8                 1000.00       5000.00                 12000        60000   
9                 1000.00       4000.00                 12000        48000   
10               25000.00      25000.00                300000       300000   
11                9166.67      27500.00                110000       330000   

    Month 1  Month 2  Month 3  Month 4  ...  Month 15  Month 16  Month 17  \
0     27917    27917    83750    83750  ...    223333    223333    223333   
1     20000    20000    60000    60000  ...    140000    140000    160000   
2      5000     5000    15000    15000  ...     35000     35000     40000   
3      3000     3000     9000     9000  ...     21000     21000     24000   
4     20833    20833    62500    62500  ...    187500    187500    187500   
5      5000     5000    15000    15000  ...     45000     45000     45000   
6      8000     8000    24000    24000  ...     72000     72000     72000   
7     10000    10000    80000    80000  ...    220000    230000    230000   
8      5000     5000    40000    40000  ...    110000    115000    115000   
9      4000     4000    32000    32000  ...     88000     92000     92000   
10    25000    25000    25000    25000  ...     25000     25000     25000   
11    27500    27500    27500    27500  ...     27500     27500     27500   

    Month 18  Month 19  Month 20  Month 21  Month 22  Month 23  Month 24  
0     251250    251250    251250    279167    279167    279167    279167  
1     160000    160000    160000    180000    180000    180000    180000  
2      40000     40000     40000     45000     45000     45000     45000  
3      24000     24000     24000     27000     27000     27000     27000  
4     187500    208333    208333    208333    229167    229167    229167  
5      45000     50000     50000     50000     55000     55000     55000  
6      72000     80000     80000     80000     88000     88000     88000  
7     240000    250000    260000    270000    270000    280000    290000  
8     120000    125000    130000    135000    135000    140000    145000  
9      96000    100000    104000    108000    108000    112000    116000  
10     25000     25000     25000     25000     25000     25000     25000  
11     27500     27500     27500     27500     27500     27500     27500  

[12 rows x 30 columns]